docs/customresourcestate-metrics.md

```diff
@@ -480,4 +480,7 @@ Examples:
 
 # if the value to be matched is a number or boolean, the value is compared as a number or boolean  
 [status, conditions, "[value=66]", name]  # status.conditions[1].name = "b"
+
+# For generally matching against a field in an object schema, use the following syntax:
+[metadata, annotations, "bar:baz"] # metadata.annotations["bar:baz"] = "baz" (if present, ignored otherwise)
```

Comment:
To be in line with how we specify values, what would you think about:
```
[metadata, annotations, "[key=bar]", "[value=baz]"]
```
I feel like it would be easier for users if we stick to the existing UX we have for values and just extend it to keys.